journey-map/server/analytics-middleware.ts

```diff
@@ -1,52 +1,59 @@
 import { RequestHandler } from "express";
 import config from "./config.js";
 
-const CACHE_TTL_MS = 60 * 1000;
-
-let cache: {
-  data: object;
-  expiresAt: number;
-} | null = null;
-
 export const fetchJourneyTransitionsHandler: RequestHandler = async (
   req,
   res,
   next,
 ) => {
   try {
-    const now = Date.now();
-    if (cache && cache.expiresAt > now) {
-      res.json(cache.data);
-      next();
-      return;
-    }
+    const filteredBody = Object.fromEntries(
+      Object.entries(req.body).filter(([key]) => key !== "environment"),
+    );
 
     const query = new URLSearchParams({
-      minutes: "30",
-      limit: "200",
+      ...filteredBody,
+      limit: "600",
     });
 
-    const response = await fetch(
-      `${config.journeyTransitionsEndpoint}?${query.toString()}`,
-      {
-        method: "POST",
-        headers: { "x-api-key": config.analyticsApiKey },
-      },
-    );
-    if (!response.ok) {
-      throw new Error(
-        `Failed to fetch journey transitions from analytics API: ${response.statusText}`,
-      );
-    }
+    const targetEnvironment = req.body
+      .environment as keyof typeof config.environment;
+    const endpoint =
+      config.environment[targetEnvironment].journeyTransitionsEndpoint;
+    const apiKey = config.environment[targetEnvironment].analyticsApiKey;
+
+    const response = await fetch(`${endpoint}?${query.toString()}`, {
```

Comment:
Might be neater to conditionally have the "?", e.g.:
```suggestion
    const response = await fetch(endpoint + query ? "?" + query.toString() : "", {
```